controllers/data.controller.js

```diff
@@ -43,669 +60,613 @@ const client = new MongoClient(uri);
+    const userScore = req.user.score;
 
-        // 실험중//////// Define filtering criteria.
-        // TODO: 유저의 티어에 따라 filter값 바뀌어야함.
-        // TODO: 이 filter로 뽑아내는 문제들이 바뀔것임. 
-        let filter;
-        if (userScore === 0) {
-            filter = {YEAR: 23};
-        } else if (userScore === 1) {
-            filter = {$and: [
-                {YEAR: {$in: [20, 22]}},
-                {LEAGUE: "LCS"}
-            ]};
+    console.log(`이게 유저 점수임 ${userScore}`);
 
-
-        } else if (userScore === 2) {
-            filter = {YEAR: {$in: [20, 22]}};
-        } else {
-            // filter = {};
-            filter = {IGN: "FAKER"};
-            
-        }
-        ///////////////////
-
-        // Find documents matching the filter
-        const cursor = coll.find(filter);
-
-        // Collect the documents in an array
-        const documents = await cursor.toArray();
-        // Send the documents as a JSON response
-        res.json(documents);
-
-
-
-        // 실험중 /////// Update user's score
-        const updatedScore = userScore + 1;
-        req.user.score = updatedScore;
-        const updatedUser = await AuthService.saveUser(req.user);
-        console.log(`이게 업데이트 이후 유저 점수임 ${req.user.score}`);
-        //
-
-
-
-    } finally {
-        await client.close();
+    // 실험중//////// Define filtering criteria.
+    // TODO: 유저의 티어에 따라 filter값 바뀌어야함.
+    // TODO: 이 filter로 뽑아내는 문제들이 바뀔것임.
+    let filter;
+    if (userScore === 0) {
+      filter = { YEAR: 23 };
+    } else if (userScore === 1) {
+      filter = { $and: [{ YEAR: { $in: [20, 22] } }, { LEAGUE: "LCS" }] };
+    } else if (userScore === 2) {
+      filter = { YEAR: { $in: [20, 22] } };
+    } else {
+      // filter = {};
+      filter = { IGN: "FAKER" };
     }
-
+    ///////////////////
+
+    // Find documents matching the filter
+    const cursor = coll.find(filter);
+
+    // Collect the documents in an array
+    const documents = await cursor.toArray();
+    // Send the documents as a JSON response
+    res.json(documents);
+
+    // 실험중 /////// Update user's score
+    const updatedScore = userScore + 1;
+    req.user.score = updatedScore;
+    const updatedUser = await AuthService.saveUser(req.user);
+    console.log(`이게 업데이트 이후 유저 점수임 ${req.user.score}`);
+    //
+  } finally {
+    await client.close();
+  }
 };
 
-
 // TODO: 이 부분을 아마 앱의 도입부로 해야할듯.
 //       티어가 " "면 일단 tier = "I4", score = 0 으로 설정하고 문제 retrieve하고 배치고사...
 //       첫번째 배치고사 이후 tier랑 score 설정 후 또 두번째 문제... 이런식으로?
 exports.setTier = async (req, res) => {
+  try {
+    await client.connect();
+
+    const db = client.db("test");
+    const coll = db.collection("users");
+
+    // Get the user's ID from the request object
+    const userID = req.user._id;
+
+    // Retrieve the user's document from the database
+    const userDocument = await coll.findOne({ _id: userID });
+
+    // Check if the "tier" field exists in the user's document
+    if (userDocument.tier === " ") {
```

Comment:
I think you can set default value for tier as I1. if (userDocument && !userDocument.tier) then you can change it to S4. So you dont need to write await coll.updateOne twice.